Source/PINDiskCache.m

```diff
@@ -540,7 +548,9 @@ - (void)trimDiskToSize:(NSUInteger)trimByteCount
 {
     [self lock];
         if (_byteCount > trimByteCount) {
-            NSArray *keysSortedBySize = [_sizes keysSortedByValueUsingSelector:@selector(compare:)];
+            NSArray *keysSortedBySize = [_metadata keysSortedByValueUsingComparator:^NSComparisonResult(PINDiskCacheMetadata * _Nonnull obj1, PINDiskCacheMetadata * _Nonnull obj2) {
```

Comment:
Interesting case I think were it would be fun to look into some algorithms that would not just take into account the byte size but also the number of accesses and the last access date. Could be a potential for an optimization for bigger files that would be trimmed very frequently but also accessed very often. Just a thought.

Reply:
Totally, would be kinda neat to have a bunch of different caching strategies that could be configured.